features/cleaner/cleaner_chunks.go

```diff
@@ -0,0 +1,149 @@
+package cleaner
+
+import (
+	"app/tools"
+	processing "app/tools/processing"
+
+	"math"
+
+	"time"
+
+	bunyan "github.com/Dewberry/paul-bunyan"
+)
+
+type chunkJob struct {
+	Filepath  string
+	Tolerance map[byte]int
+	Offset    tools.OrderedPair
+	ICP       innerChunkPartition
+	Size      tools.OrderedPair
+	AreaSize  float64
+	AdjType   int
+}
+
+type chunkFillStruct struct {
+	AreaMap [][]square
+	cStats  cleanerStats
+	Offset  tools.OrderedPair
+}
+
+type innerChunkPartition struct {
+	RStart int
+	REnd   int
+	CStart int
+	CEnd   int
+}
+
+func sliceAreaMap(areaMap *[][]square, ICP innerChunkPartition) {
+	*areaMap = (*areaMap)[ICP.RStart:ICP.REnd]
+	for r := 0; r < len(*areaMap); r++ {
+		(*areaMap)[r] = (*areaMap)[r][ICP.CStart:ICP.CEnd]
+	}
+}
+
+func cleanChunk(jobs chan chunkJob, chunkChannel chan chunkFillStruct) {
+	for j := range jobs {
+		areaMap, err := readFileChunk(j.Filepath, tools.MakePair(j.Offset.R-j.ICP.RStart, j.Offset.C-j.ICP.CStart), j.Size)
+		if err != nil {
+			bunyan.Fatal(err)
+		}
+
+		cStats := cleanAreaMap(&areaMap, j.Tolerance, j.AreaSize, j.AdjType, j.ICP)
+		sliceAreaMap(&areaMap, j.ICP)
+
+		chunkChannel <- chunkFillStruct{areaMap, cStats, j.Offset}
+	}
+
+}
+
+func bufferSize(adjType int, tolerance map[byte]int) tools.OrderedPair {
+	maxTolerance := tools.Max(tolerance[byte(0)], tolerance[byte(1)])
+	switch adjType {
+	case 4:
+		return tools.MakePair(maxTolerance, maxTolerance)
+
+	default: // case 8
+		return tools.MakePair(maxTolerance*2, maxTolerance*2)
+	}
+}
+
+//Return start of chunk, size of chunk (both Ordered Pairs), valid
+func makeChunk(buffer tools.OrderedPair, chunkSize tools.OrderedPair, rowsAndCols tools.OrderedPair, r int, c int) (innerChunkPartition, tools.OrderedPair) {
+	startBuffer := tools.MakePair(tools.Max(0, r-buffer.R), tools.Max(0, c-buffer.C))
+	endBuffer := tools.MakePair(tools.Min(rowsAndCols.R, r+chunkSize.R+buffer.R), tools.Min(rowsAndCols.C, c+chunkSize.C+buffer.C))
+	size := tools.MakePair(endBuffer.R-startBuffer.R, endBuffer.C-startBuffer.C)
+
+	newR := tools.Min(buffer.R, r)
+	newC := tools.Min(buffer.C, c)
+	innerChunk := innerChunkPartition{newR, tools.Min(newR+chunkSize.R, size.R), newC, tools.Min(newC+chunkSize.C, size.C)}
+
+	return innerChunk, size
+}
+
+func CleanWithChunking(filepath string, outfile string, toleranceIsland float64, toleranceVoid float64, chunkSize tools.OrderedPair, adjType int) error {
+	gdal, rowsAndCols, err := processing.GetInfoGDAL(filepath)
+	bunyan.Infof("img rows x cols: %v", rowsAndCols)
+	if err != nil {
+		return err
```

Comment:
Same question here regarding error logging. Guessing it is happening outside the function but we should ensure we capture all errors in the log.